change Clients to a public field on watcher

eikenb opened the pull request:
Not really internal and handy to have access to for other things.

Specifically it makes it easier to implement a custom Recaller for
testing template validity in CTS (example of this to be added to
doc_test.go).

eikenb commented:
@mkam .. here's that change to make the watcher.Clients public.

lornasong reviewed (approved):
makes sense to me!

mkam commented:
Went to try this out and I ran into a complication, which is we use a [Watcher interface](https://github.com/hashicorp/consul-terraform-sync/blob/main/templates/hcat.go#L42) in CTS, so I still wasn't able to access the clients easily. Could we also have a getter method for the clients?

eikenb commented:
Yeah... didn't think of that. 

I played with the idea of turning the ClientSet into a singleton with a module level access point but couldn't convince myself it was a good idea. So I'll convert the public Clients attribute to a public Clients() method.

eikenb commented:
@mkam... I've updated this PR with a version using a getter. I also added the getter to the Watcherer interface. Seems like that is all you need. Please let me know if there is anything else. Thanks.

lornasong reviewed (approved):
`Clients()` method looks good to me!

mkam reviewed (approved):
Looks good to me too, thanks for making the adjustments!
